docs/basics/events.md

```diff
@@ -62,32 +62,104 @@ for an elaborate example which uses events.
 
 ## Event Definition
 
+Since `ink!` version `5.0`, events can be defined independently of the contract which emits them. 
+Events can now be defined once and shared across multiple contracts. 
+
+This is useful for events for contracts which conform to standards such as `ERC20`: 
+contract indexers/explorers are now able to group all e.g. `Transferred` events.
+
 This is how an event definition looks:
 
 ```rust
-#[ink(event)]
+use ink::primitives::AccountId;
+
+#[ink::event]
 pub struct Transferred {
     #[ink(topic)]
     from: Option<AccountId>,
-
     #[ink(topic)]
     to: Option<AccountId>,
+    amount: u128,
+}
+```
+> Note that generics are [not currently supported](https://github.com/paritytech/ink/issues/2044)
+> , so the concrete types of `Environment` 
+> specific types such as `AccountId` must match up with the types used in the contract.
 
-    amount: Balance
+This definition can exist within a contract definition module (inline events), in a different 
+module in the same crate or even in a different crate to be shared by multiple contracts.
 
-}
+### Legacy syntax for inline Event definitions
+
+Events defined within a `#[ink::contract]` module can continue to use the original syntax for an 
+event definition, using the `#[ink(event)]` attribute. Under the covers this is simply expanded 
+to the new top level `#[ink::event]` macro, so both events defined using the legacy style and 
+using the new `event` attribute macro directly will behave exactly the same.
+
+### Topics
+
+When an event is emitted, 0 or more topics can be associated with it. The event is then indexed 
+together with other events with the same topic value.
+
+An event's fields can be annotated with `#[ink(topic)]` (see example), which will result in a 
+topic derived from the value of that field being emitted together with the event.
+
+Topics are by default a 32 byte array (`[u8; 32]`), although this is configurable on the 
```

Comment:
Is it actually configurable? I thought ink is hardcoded to use blake 256 still

Reply:
The type of the resulting topic is configurable yes. So if it is shorter I guess it will be truncated and longer padded.

https://github.com/paritytech/ink/blob/27407b5f59f24233cddc3cea59cc2213ba2c6674/crates/env/src/engine/on_chain/impls.rs#L157

Our way of generating the topic value itself is hardcoded to blake2.